test-case-reporting/index.ts

```diff
@@ -16,23 +16,72 @@
 
 require('dotenv').config();
 
-import {KarmaReporter} from 'test-case-reporting';
+import {KarmaReporter, TestRun} from 'test-case-reporting';
+import {TestResultRecord} from './src/test_result_record';
+import {dbConnect} from './src/db';
 import express from 'express';
 const app = express();
 const PORT = process.env.PORT || 8080;
+const MAX_PAGE_SIZE = 500;
+
+function renderTestRunList(testRuns: Array<TestRun>): string {
+  return `Rendering ${testRuns.length} test runs!`;
+}
 
 app.get('/', (req, res) => {
   res.send('Hello, world, 2.0!');
 });
 
-app.get('/test-results/pr/:prNumber', (req, res) => {
-  const {prNumber} = req.params;
-  res.send(`List of test cases for PR number ${prNumber}`);
+app.get('/test-results/build/:buildNumber', async (req, res) => {
+  const {buildNumber} = req.params;
+  const {json} = req.query;
+  const db = dbConnect();
+  const testResultRecord = new TestResultRecord(db);
+
+  const pageSize = 100;
+
+  const testRunJson = {
+    testRuns: await testResultRecord.getTestRunsOfBuild(buildNumber, {
+      limit: pageSize,
+      offset: 0,
+    }),
+  };
+
+  if (json) {
+    res.json(testRunJson);
+  } else {
+    res.send(renderTestRunList(testRunJson.testRuns));
+  }
 });
 
-app.get('/test-results/history/:testCaseId', (req, res) => {
+app.get('/test-results/history/:testCaseId', async (req, res) => {
   const {testCaseId} = req.params;
-  res.send(`Test history for test with name/ID ${testCaseId}`);
+  const {json} = req.query;
+  const {limit, offset} = req.query;
+  const db = dbConnect();
+  const testResultRecord = new TestResultRecord(db);
+
+  let limitNum = parseInt(limit.toString(), 10);
+  const offsetNum = parseInt(offset.toString(), 10);
+  if (limitNum > MAX_PAGE_SIZE) {
+    limitNum = MAX_PAGE_SIZE;
+    console.warn(
+      `Maximum query size exceeded. Showing only first ${MAX_PAGE_SIZE} results.`
+    );
+  }
```

Comment:
It may be useful to have a shared `extractPageInfo(req)` helper used by both of the list queries (and any future ones)

Reply:
Good idea! Done.